File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Partner.kt
package vn.hitechcloud.sdk

/** Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates */
class PartnerResource(private val httpClient: HttpClient) {
    /** GET /api/partner - Get partner info */
    fun getPartnerInfo(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner", params)
    }

    /** POST /api/partner/apply - Apply to partner program */
    fun apply(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/partner/apply", params)
    }

    /** GET /api/partner/tiers - List partner tiers */
    fun listTiers(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/tiers", params)
    }

    /** GET /api/partner/pricing - Get partner pricing */
    fun getPricing(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/pricing", params)
    }

    /** GET /api/partner/customers - List customers */
    fun listCustomers(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/customers", params)
    }

    /** GET /api/partner/earnings - Get earnings */
    fun getEarnings(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/earnings", params)
    }

    /** GET /api/partner/wallet - Get wallet balance */
    fun getWallet(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/wallet", params)
    }

    /** GET /api/partner/payouts - List payouts */
    fun listPayouts(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/payouts", params)
    }

    /** POST /api/partner/payouts - Request payout */
    fun requestPayout(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/partner/payouts", params)
    }

    /** GET /api/partner/leads - List leads */
    fun listLeads(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/leads", params)
    }

    /** POST /api/partner/leads - Create lead */
    fun createLead(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/partner/leads", params)
    }

    /** GET /api/partner/referral - Get referral info */
    fun getReferral(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/referral", params)
    }

    /** GET /api/partner/rates - Get exchange rates */
    fun getRates(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/partner/rates", params)
    }

}

File: sdk/swift/Sources/HiTechCloudSDK/Partner.swift
import Foundation

/// Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates
public class PartnerResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/partner - Get partner info
    public func getPartnerInfo(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner", params: params)
    }

    /// POST /api/partner/apply - Apply to partner program
    public func apply(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/partner/apply", params: params)
    }

    /// GET /api/partner/tiers - List partner tiers
    public func listTiers(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/tiers", params: params)
    }

    /// GET /api/partner/pricing - Get partner pricing
    public func getPricing(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/pricing", params: params)
    }

    /// GET /api/partner/customers - List customers
    public func listCustomers(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/customers", params: params)
    }

    /// GET /api/partner/earnings - Get earnings
    public func getEarnings(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/earnings", params: params)
    }

    /// GET /api/partner/wallet - Get wallet balance
    public func getWallet(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/wallet", params: params)
    }

    /// GET /api/partner/payouts - List payouts
    public func listPayouts(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/payouts", params: params)
    }

    /// POST /api/partner/payouts - Request payout
    public func requestPayout(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/partner/payouts", params: params)
    }

    /// GET /api/partner/leads - List leads
    public func listLeads(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/leads", params: params)
    }

    /// POST /api/partner/leads - Create lead
    public func createLead(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/partner/leads", params: params)
    }

    /// GET /api/partner/referral - Get referral info
    public func getReferral(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/referral", params: params)
    }

    /// GET /api/partner/rates - Get exchange rates
    public func getRates(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/partner/rates", params: params)
    }

}
